### PR TITLE
add test cleanup/state restore for certain tests; add test tags

### PR DESCRIPTION
**Pull request by richm**
Add test state save/state restore for cleanup for certain tests
that we run on a single machine in CI.  Add tags for other tests
that we use in CI.

### CODE REVIEW

File: tests/playbooks/tests_802_1x.yml [hunk 1]
```diff
@@ -122,3 +127,8 @@
               command: update-ca-trust
           tags:
             - "tests::cleanup"
+
+- name: Restore host state
```
Comment (tyll) [inline tests/playbooks/tests_802_1x.yml:131]: For other cleanup tasks, we added them in an always block to ensure that they always run. Do we need this here, too?

File: tests/playbooks/tests_bond.yml [hunk 2]
```diff
@@ -8,6 +13,8 @@
     dhcp_interface1: test1
     port2_profile: bond0.1
     dhcp_interface2: test2
+  tags:
+    - "tests::expfail"
```
Comment (tyll) [inline tests/playbooks/tests_bond.yml:17]: Why is this expected to fail? Would be good to explain/link to an issue/BZ to track this.

File: tests/tasks/restore_state.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    path: /etc/sysconfig/network
+    state: absent
+  when:
+    - etc_sysconfig_network_stat is defined
+    - not etc_sysconfig_network_stat.stat.exists
+
+- name: Restore services
+  import_tasks: restore_services_state.yml
+
+- name: reload NetworkManager
```
Comment (tyll) [inline tests/tasks/restore_state.yml:15]: Is it a problem when NetworkManager or other services are installed during the test and not removed/disabled afterwards?

**Comment by jharuda** [comment]
[citest]

**Comment by jharuda** [comment]
[citest]

**Pull request closed without merging**